resolver/serviceResolver.go

```diff
@@ -411,7 +433,43 @@ func (resolver *serviceResolver) validateTxRequestReturningGuardian(
 	}
 
 	// only get the guardian for first tx, as all of them must have the same one
-	guardianInfo, err := resolver.getGuardianForTx(txs[0], userInfo)
+	guardianInfo, err := resolver.getGuardianInfoFromAddress(txs[0].GuardianAddr, userInfo)
+	if err != nil {
+		return core.GuardianInfo{}, otpVerifyCodeData, err
+	}
+
+	return guardianInfo, otpVerifyCodeData, nil
+}
+
+func (resolver *serviceResolver) validateMsgRequestReturningGuardian(
+	userAddress sdkCore.AddressHandler,
+	guardianAddr string,
+	userIp,
+	code,
+	secondCode string) (core.GuardianInfo, *requests.OTPCodeVerifyData, error) {
```

Comment:
```suggestion
	secondCode string,
) (core.GuardianInfo, *requests.OTPCodeVerifyData, error) {
```

also, I would rename this method to `verifyCodesReturningGuardian` and reuse it as part of `validateTxRequestReturningGuardian` ( L422-L446 is duplicated with L465-L482 )